src/blr/Snakefile

```diff
@@ -247,15 +247,80 @@ rule symlink_reference_variants:
         "ln -s {config[reference_variants]} {output}"
 
 
-rule call_variants_freebayes:
+rule add_RG_tag:
     output:
-         vcf = "variants.called.vcf"
+        bam = "mapped.sorted.tag.mkdup.bcmerge.mol.filt.RG.bam",
+        bai = "mapped.sorted.tag.mkdup.bcmerge.mol.filt.RG.bam.bai"
     input:
-         bam = "mapped.sorted.tag.mkdup.bcmerge.mol.filt.bam"
-    log: "call_variants_freebayes.log"
+        bam = "mapped.sorted.tag.mkdup.bcmerge.mol.filt.bam"
+    log: "add_RG_tag.log"
     shell:
-         "freebayes"
-         " -f {config[genome_reference]}"
-         " {input.bam} |"
-         "vcffilter -f 'AC > 0' 1>"
-         " {output.vcf} 2> {log}"
+        "samtools addreplacerg"
```

Comment:
This is good but i still think we should move this to another rule to limit read/writing of files. My suggestion would be to stream the output from `filterclusters` through this command, something like...
```
blr filterclusters input.bam ... | samtools addreplacerg - ... | tee out.bam | samtools index. 
```
This also means one less step of indexing.

Reply:
Yeah that's right, forgot about that. Makes total since this should more or less be only read/write and if it ever comes close to be CPU throttling we can add threading to it. 

I added it even further up in the tagbam rule, thinking that seemed like a good rule name for incorporation of any number of SAM tags. 

Possibly it even makes sense to pipe this directly from the mapper (since at least broad institute software assume you set RG tags for all libraries). However, I opted not to do that for now since it's already quite a large piping operation, thinking this could is more readable.